core/src/main/scala/cats/mtl/MonadChronicle.scala

```diff
@@ -0,0 +1,29 @@
+package cats
+package mtl
+
+import cats.data.Ior
+
+trait MonadChronicle[F[_], E] extends Serializable {
+  val monad: Monad[F]
+
+  def dictate(c: E): F[Unit]
+
+  def disclose[A](c: E)(implicit M: Monoid[A]): F[A]
+
+  def confess[A](c: E): F[A]
+
+  def memento[A](ma: F[A]): F[Either[E, A]]
+
+  def absolve[A](a: A, ma: F[A]): F[A]
+
+  def condemn[A](ma: F[A]): F[A]
+
+  def retcon[A](cc: E => E, ma: F[A]): F[A]
```

Comment:
With memento returning Ior this could be:

```scala
def retcon[A](cc: E => E, ma: F[A]): F[A] =
  memento(ma).map {
    case Left(e) => confess(cc(e))
    case Right(a) => pure(a)
    case Both(e, a) => dictate(cc(e)).as(a)
  }
```